File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/lobby/PlayerChangedToSpectator.java
package tech.zolhungaj.amqapi.servercommands.gameroom.lobby;

import tech.zolhungaj.amqapi.servercommands.CommandType;
import tech.zolhungaj.amqapi.servercommands.objects.PlayerIdentifier;

@CommandType("Player Changed To Spectator")
public record PlayerChangedToSpectator(
        PlayerIdentifier spectatorDescription,
        PlayerIdentifier playerDescription,
        Boolean isHost
){}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/lobby/PlayerChangedToSpectator.kt
package tech.zolhungaj.amqapi.servercommands.gameroom.lobby

import tech.zolhungaj.amqapi.servercommands.CommandType
import tech.zolhungaj.amqapi.servercommands.objects.PlayerIdentifier

@JvmRecord
@CommandType("Player Changed To Spectator")
data class PlayerChangedToSpectator(
    val spectatorDescription: PlayerIdentifier,
    val playerDescription: PlayerIdentifier,
    val isHost: Boolean
)
